### ISSUE TITLE
Synonym token types differ between Solr/Wordnet format

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#44507]
ES version tested: 7.2.0

I noticed a subtle difference in the way tokens are analyzed when playing around with Wordnet format synonym files vs. the way they are treated when using the Solr format. A simple example:

Using the Solr format:
```
PUT /test_index2
{
    "settings": {
        "index" : {
            "analysis" : {
                "analyzer" : {
                    "synonym" : {
                        "tokenizer" : "whitespace",
                        "filter" : ["synonym"]
                    }
                },
                "filter" : {
                    "synonym" : {
                        "type" : "synonym",
                        "synonyms" : [ "one, uno, eins" ]
                    }
                }
            }
        }
    }
}

GET /test_index2/_analyze
{
  "analyzer": "synonym",
  "text" : "uno"
}
```
produces
```
{
  "tokens" : [
    {
      "token" : "uno",
      "start_offset" : 0,
      "end_offset" : 3,
      "type" : "word",   // <- expected since thats the original term in the input text
      "position" : 0
    },
    {
      "token" : "one",
      "start_offset" : 0,
      "end_offset" : 3,
      "type" : "SYNONYM",
      "position" : 0
    },
    {
      "token" : "eins",
      "start_offset" : 0,
      "end_offset" : 3,
      "type" : "SYNONYM",
      "position" : 0
    }
  ]
}
```

Using what I think should be the same rules using the Wordnet format:
```
PUT /test_index3
{
    "settings": {
        "index" : {
            "analysis" : {
                "analyzer" : {
                    "synonym" : {
                        "tokenizer" : "whitespace",
                        "filter" : ["synonym"]
                    }
                },
                "filter" : {
                    "synonym" : {
                        "type" : "synonym",
                        "synonyms" : [ 
                            "s(100000001,1,'one',v,1,0).",
                            "s(100000001,2,'uno',v,1,0).",
                            "s(100000001,3,'eins',v,1,0)."
                          ],
                        "format" : "wordnet"
                    }
                }
            }
        }
    }
}

GET /test_index3/_analyze
{
  "analyzer": "synonym",
  "text" : "uno"
}
```

produces

```
{
  "tokens" : [
    {
      "token" : "one",
      "start_offset" : 0,
      "end_offset" : 3,
      "type" : "SYNONYM",  
      "position" : 0
    },
    {
      "token" : "uno",
      "start_offset" : 0,
      "end_offset" : 3,
      "type" : "SYNONYM",  // <- that was type "word" and in the first position in the other case
      "position" : 0
    },
    {
      "token" : "eins",
      "start_offset" : 0,
      "end_offset" : 3,
      "type" : "SYNONYM",
      "position" : 0
    }
  ]
}
```

I don't think we use anything else than the synset-id (first number) and the term itself from the wordnet format, so the rules should be equivalent. I don't think the different token type changes anything in terms of how terms are indexed (might be different term statistics though?) or searched, but wondering what works differently here and if it matters.

**Comment by cbuescher** [comment]
After some testing, this seems to boil down to small differences in how WordnetSynonymParser and SolrSynonymParser work. When in expand mode, the former [adds all combinations of a synset to the Synonym map](https://github.com/apache/lucene-solr/blob/master/lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/WordnetSynonymParser.java#L98), including the input and output term being the same. This is different in the SolrSynonymParser where things are filtered out [when input/ouput are referring to the same term](https://github.com/apache/lucene-solr/blob/master/lucene/analysis/common/src/java/org/apache/lucene/analysis/synonym/SolrSynonymParser.java#L120).

**Comment by cbuescher** [comment]
I opened https://github.com/apache/lucene-solr/pull/981 that should fix this issue.

**Comment by romseygeek** [comment]
LUCENE-9030 will be included lucene 8.4, which will be released fairly soon.  Our master branch is already running on an 8.4 snapshot, so I think we can close this issue?

**Comment by cbuescher** [comment]
> I think we can close this issue?

Agreed.
